pkg/binlog/position.go

```diff
@@ -132,3 +132,26 @@ func verifyUUIDSuffix(suffix string) bool {
 	}
 	return true
 }
+
+// AdjustPosition adjusts the filename with uuid suffix in mysql position
+// for example: mysql-bin|000001.000002 -> mysql-bin.000002
+func AdjustPosition(pos gmysql.Position) gmysql.Position {
+	realPos, err := RealMySQLPos(pos)
+	if err != nil {
+		// just return the origin pos
+		return pos
+	}
+
+	return realPos
+}
+
+// ComparePosition returns:
+//   1 if pos1 is bigger than pos2
+//   0 if pos1 is equal to pos2
+//   -1 if pos1 is less than pos2
+func ComparePosition(pos1, pos2 gmysql.Position) int {
+	pos1 = AdjustPosition(pos1)
+	pos2 = AdjustPosition(pos2)
```

Comment:
If the UUID suffix exists for both of them, should we need to compare it? or can we still support upstream master-slave switch if in _local_ mode?

Reply:
it is a bug, I fix it in a141b61